File: validate_json.py
import json
import sys

file_path = r'c:\website\website\eco-tech\server\data\members.json'


try:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
        
        # Check for smart quotes
        smart_quotes = ['\u201c', '\u201d', '\u2018', '\u2019']
        found_smart = False
        for i, char in enumerate(content):
            if char in smart_quotes:
                lines = content[:i].splitlines()
                line_no = len(lines)
                print(f"Found smart quote at line {line_no}: {char!r}")
                found_smart = True
                
        if not found_smart:
             try:
                json.loads(content)
                print("JSON is valid (no smart quotes found)")
             except json.JSONDecodeError as e:
                print(f"JSON error: {e}")

except Exception as e:
    print(f"File validation error: {e}")

File: validate_json.js
const fs = require('fs');
const path = 'c:\\website\\website\\eco-tech\\server\\data\\members.json';

try {
    const data = fs.readFileSync(path, 'utf8');
    JSON.parse(data);
    console.log('JSON is valid');
} catch (e) {
    console.log('JSON error:', e.message);
    if (e.message.includes('position')) {
        const pos = parseInt(e.message.match(/position (\d+)/)[1]);
        const lines = fs.readFileSync(path, 'utf8').substring(0, pos).split('\n');
        console.log('Error at line:', lines.length, 'column:', lines[lines.length - 1].length);
    }
}
